### PR TITLE
[iceberg] introduce procedure for migrating iceberg table to paimon

### PR DESCRIPTION
**Pull request by LsomeYeah**
### Purpose

Linked issue: close #xxx

In pr #4639, we had introduced the feature to migrating iceberg table to paimon.
This pr added corresponding procedures and IT cases.

### Tests

org.apache.paimon.hive.procedure.MigrateIcebergTableProcedureITCase
org.apache.paimon.procedure.MigrateIcebergTableProcedure23ITCase
org.apache.paimon.procedure.MigrateIcebergTableProcedure31ITCase

### CODE REVIEW

File: paimon-core/src/main/java/org/apache/paimon/iceberg/IcebergMigrator.java [hunk 1]
```diff
@@ -0,0 +1,347 @@
...
+import java.util.Map;
+import java.util.concurrent.Callable;
+import java.util.concurrent.ConcurrentHashMap;
+import java.util.concurrent.Future;
+import java.util.concurrent.ThreadPoolExecutor;
+import java.util.stream.Collectors;
+
+import static org.apache.paimon.utils.ThreadPoolUtils.createCachedThreadPool;
+
+/** migrate hive table to paimon table. */
```
Comment (xuzifu666) [inline paimon-core/src/main/java/org/apache/paimon/iceberg/IcebergMigrator.java:61]: migrate iceberg table to paimon table？

File: paimon-flink/paimon-flink-common/src/main/java/org/apache/paimon/flink/procedure/MigrateTableProcedure.java [hunk 2]
```diff
@@ -20,20 +20,18 @@
...
 
 import org.apache.flink.table.annotation.ArgumentHint;
 import org.apache.flink.table.annotation.DataTypeHint;
 import org.apache.flink.table.annotation.ProcedureHint;
 import org.apache.flink.table.procedure.ProcedureContext;
-import org.slf4j.Logger;
-import org.slf4j.LoggerFactory;
 
 /** Migrate procedure to migrate hive table to paimon table. */
 public class MigrateTableProcedure extends ProcedureBase {
```
Comment (xuzifu666) [inline paimon-flink/paimon-flink-common/src/main/java/org/apache/paimon/flink/procedure/MigrateTableProcedure.java:33]: Also implement in spark side?

**Review by xuzifu666 — COMMENTED** [review]
Good job! It is a important way for iceberg user migrate to paimon, left some comments @LsomeYeah

File: paimon-flink/paimon-flink-common/src/main/java/org/apache/paimon/flink/procedure/MigrateTableProcedure.java (hunk 2 above)
Comment (LsomeYeah) [reply paimon-flink/paimon-flink-common/src/main/java/org/apache/paimon/flink/procedure/MigrateTableProcedure.java:33]: Will add it later.

**Pull request closed without merging**